openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+                scatter.get_xs(nuclides=nuclide, xs_type=xs_type,
+                               subdomains=subdomain)
+
+    def set_multiplicity_matrix_mgxs(self, nuscatter, scatter=None,
+                                     temperature=294., nuclide='total',
+                                     xs_type='macro', subdomain=None):
+        """This method allows for either the direct use of only an
+        openmc.mgxs.MultiplicityMatrixXS or an openmc.mgxs.ScatterMatrixXS and
+        openmc.mgxs.ScatterMatrixXS to be used to set the scattering
+        multiplicity for this XSdata object. Multiplicity, in OpenMC parlance,
+        is a factor used to account for the production of neutrons introduced by
+        scattering multiplication reactions, i.e., (n,xn) events. In this sense,
+        the multiplication matrix is simply defined as the ratio of the
+        nu-scatter and scatter matrices.
+
+        Parameters
+        ----------
+        nuscatter: openmc.mgxs.ScatterMatrixXS or openmc.mgxs.MultiplicityMatrixXS
+            MGXS Object containing the matrix cross section for the domain
+            of interest.
+        scatter: openmc.mgxs.ScatterMatrixXS
+            MGXS Object containing the scattering matrix cross section
+            for the domain of interest.
+        temperature : float
+            Temperature (in Kelvin) of the data. Defaults to room temperature
+            (294K).
+        nuclide : str
+            Individual nuclide (or 'total' if obtaining material-wise data)
+            to gather data for.  Defaults to 'total'.
+        xs_type: {'macro', 'micro'}
+            Provide the macro or micro cross section in units of cm^-1 or
+            barns. Defaults to 'macro'.
+        subdomain : iterable of int
+            If the MGXS contains a mesh domain type, the subdomain parameter
+            specifies which mesh cell (i.e., [i, j, k] index) to use.
+
+        See also
+        --------
+        openmc.mgxs.Library.create_mg_library()
+        openmc.mgxs.Library.get_xsdata()
+
+        """
+
+        check_type('nuscatter', nuscatter, (openmc.mgxs.ScatterMatrixXS,
+                                            openmc.mgxs.MultiplicityMatrixXS))
+        check_value('energy_groups', nuscatter.energy_groups,
+                    [self.energy_groups])
+        check_value('domain_type', nuscatter.domain_type,
+                    openmc.mgxs.DOMAIN_TYPES)
+        check_type('temperature', temperature, Real)
+        check_value('temperature', temperature, self.temperatures)
+
+        if scatter is not None:
+            check_type('scatter', scatter, openmc.mgxs.ScatterMatrixXS)
+            if isinstance(nuscatter, openmc.mgxs.MultiplicityMatrixXS):
+                msg = 'Either an MultiplicityMatrixXS object must be passed ' \
+                      'for "nuscatter" or the "scatter" argument must be ' \
+                      'provided.'
+                raise ValueError(msg)
+       
```

Comment:
Since `y` is only used once, it seems like the values could be generated in a one-off basis inside of the ensuing for-loop instead of using a comprehension.

```
for l in range(xsdata.num_orders):
     coefficient = interp1d(mu_self, orig_data)(mu_fine) * eval_legendre(l, mu_fine)
     new_data[..., l] = simps( coefficient , mu_fine)
```

`coefficient` may not be the best variable name, so perhaps replace with something more appropriate.